extensions-contrib/materialized-view-maintenance/src/main/java/io/druid/indexing/materializedview/MaterializedViewMetadataCoordinator.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.indexing.materializedview;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Supplier;
+import com.google.common.base.Throwables;
+import com.google.common.collect.Maps;
+import com.google.common.hash.Hashing;
+import com.google.common.io.BaseEncoding;
+import com.google.inject.Inject;
+import io.druid.indexing.overlord.DataSourceMetadata;
+import io.druid.java.util.common.DateTimes;
+import io.druid.java.util.common.Pair;
+import io.druid.java.util.common.StringUtils;
+import io.druid.metadata.MetadataStorageTablesConfig;
+import io.druid.metadata.SQLMetadataConnector;
+import io.druid.timeline.DataSegment;
+import org.joda.time.Interval;
+import org.skife.jdbi.v2.Handle;
+import org.skife.jdbi.v2.StatementContext;
+import org.skife.jdbi.v2.tweak.HandleCallback;
+import org.skife.jdbi.v2.tweak.ResultSetMapper;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Map;
+
+public class MaterializedViewMetadataCoordinator 
+{
+  
+  private final Supplier<MetadataStorageTablesConfig> dbTables;
+  private final SQLMetadataConnector connector;
+  private final ObjectMapper objectMapper;
+
+  @Inject
+  public MaterializedViewMetadataCoordinator(
+      ObjectMapper objectMapper,
+      Supplier<MetadataStorageTablesConfig> dbTables,
+      SQLMetadataConnector connector
+  )
+  {
+    this.objectMapper = objectMapper;
+    this.dbTables = dbTables;
+    this.connector = connector;
+  }
+  
+  public void insertDataSourceMetadata(String dataSource, DataSourceMetadata metadata) 
```

Comment:
Yes, this method can be merged into `IndexerSQLMetadataStorageCoordinator.resetDataSourceMetadata()`. However, maybe we can do it in another pr, because we should consider the logic of code where used this method.